### DISCUSSION TITLE
Code of Conduct Committee Description and Composition

### DISCUSSION THREAD

**Discussion by cblecker** [cncf/wg-coc#17] [Request for Feedback]
The working group is seeking feedback on the description and composition of the CNCF Code of Conduct Conduct Committee this working group will be creating.

This includes but is not limited to:
- The committee's composition
- Eligibility criteria
- Election/appointment procedures
- Term lengths and limits
- How the committee makes decisions
- How the committee accesses external resources

An initial draft has been created here: https://github.com/cncf/wg-coc/blob/main/coc-committee-description-proposed.md

If you have feedback on this document, please leave your comments in this discussion. Once consensus around points is reached, then PRs will be created to update the draft.

**Comment by riecatnor** [comment]
Hi @cblecker - My first impression after reviewing the CNCF Code of Conduct Committee documentation is that it might need a bit more information. Reviewing the Kubernetes Code of Conduct Committee documentation [0], I see they have a Charter [1]. Is there a plan to have something similar or will those topics be included in this document? I think having it all in one place or two separate documents is fine, just a matter of what is the best fit for this instance. 

**Purpose**
- This section is very straightforward as it stands. Although I think it does cover the basics of what the purpose of the CoCC is, I think it might be missing the underlying purpose: creating and stewarding a safe and inclusive environment for the CNCF community. I think the Kubernetes Charter [1] has a great example of a mission/purpose statement. Maybe that could be adapted for this section?
- The paragraph about the relationship to the Linux Foundation might make more sense under the Members section? From my perspective, it doesn’t directly link to the purpose of the CoCC, it feels like more of a supporting piece of information around the reason LF staff will be a part of the committee.

**Members**
- This part LGTM

**How CoC Committee Members Are Selected**
- I see the Kubernetes CoCC has documentation for elections [2], using a nomination process, would this be appropriate and could it be adapted for the CNCF CoCC? Other communities are doing it differently: appointing CoCC members, or even having folks self nominate more casually by email(and the current CoCC voting on their inclusion). There’s a range of ways to approach this, and I am not sure exactly what will fit best with CNCF culture. 

**How Decisions Are Made**
- My first impression is this is a very dense paragraph and might benefit from being broken up a bit for readability. The text overall seems good. 

**External Support Resources**
- I am wondering if it would be appropriate to add “legal representation” in this list? Although hopefully that wouldn’t be a common situation, it seems like the place to add it, if applicable to the CNCF CoCC. 

**Incident Resolution Procedures**
- I plan to review the corresponding documentation and will comment on the discussion thread for that doc.

[0] https://github.com/kubernetes/community/tree/master/committee-code-of-conduct
[1] https://github.com/kubernetes/community/blob/master/committee-code-of-conduct/charter.md
[2] https://github.com/kubernetes/community/blob/master/committee-code-of-conduct/election.md

**Comment by TheFoxAtWork** [comment]
Cross referencing to Brandon's points (@lumjjb), especially on confidentiality of reporting and the qualities of individuals serving the community in this capacity and the response i provided on assumptive qualities of those individuals:
> if the reporter is not aware of conflicts of interest, members with hard conflicts would see confidential information about the full report.

> Those assumptive qualities [of the code of conduct members] should be defined publicly such that reporters feel comfortable - even with a hard conflict from a committee member - that what they report is protected, only used for the purpose it is intended, and does not further another's ends or ambitions. Some assumptive qualities that come to my mind (things i strive toward): trustworthiness, compassion, kindness, fairness, broader perspective than themselves, impartiality, accountability, and due consideration.

**Comment by tpepper** [comment]
I wouldn't say we have an airtight model yet, but the Kubernetes CoCC and Steering Committee have been working on how to establish a plan for if/when an individual needs removed or a body has resigned as a whole, fell into inactivity, or must otherwise be removed and restablished for the community good.  Would be beneficial to consider these scenarios also during the drafting here for CNCF CoCC.

**Comment by endocrimes** [comment]
One potential concern with the makeup of the committee here is that with an even number of members a majority isn't forced in any vote. This has some advantages (in that the "default" contested decision isn't immediately resolved and needs further debate) - but it _does_ mean that the committee can end up unable to progress.

**Comment by cblecker** [comment]
Here's some additional feedback I've received on this topic:

Membership:
- The current proposed makeup of the CoCC focuses on representation as opposed to identifying willing, educated, and able participants. The groups highlighted (TOC, TAG-CS, Maintainers) are all busy folks with existing responsibilities, and restricting eligibility to serve based on those groups may not be optimal as far as the significant time commitment serving on a CoCC can be.
- Additionally, restricting eligibility to serve on the committee to folks who are in active community leadership roles already, increases the chance of conflicts of interest.
- An alternative might be looking at the Kubernetes CoCC model, where anyone is eligible to serve, but they need to be elected by the leadership group: https://github.com/kubernetes/community/blob/master/committee-code-of-conduct/election.md

Alternates:
- There are references to Governing Board alternates? Unclear what that is referring to
- Under what circumstances would an alternate serve on an ongoing basis, as opposed to just electing a new primary member?
- Are CNCF alternates subject to the same rules as the other alternates (not involved in every incident, etc)?

Liaisons:
- Liaisons should be folks who aid the CoCC, as opposed to someone who can be appointed/injected into the CoCC (in particular in the case of the LF Liaison). It should be up to the CoCC if they invite a liaison into a particular incident discussion, and they should have say on who the liaison is if that liaison has a potential conflict of interest.

**Reply by TheFoxAtWork** [reply]
RE: Membership - Excellent call out. i think there is a good opportunity to strike a harmonious balance and the Kubernetes CoCC structure could be leveraged as part of achieving this.

**Reply by joannalee333** [reply]
@cblecker, I modified the PR to incorporate your comments above and the comments from the Co-Chairs

**Comment by cblecker** [comment]
Another piece of feedback I received: Do we want to have some kind of maximal representation clause reflected so that no more than X number of members of the CoCC are from the same employer?

**Reply by tpepper** [reply]
Good catch.  I don't know why I was thinking it is there, but no!  It's somewhat implicit in the how decisions are made and conflicts-of-interest texts, but really ought to be explicit.

**Reply by TheFoxAtWork** [reply]
Concur - @joannalee333 lets get that added.

**Reply by joannalee333** [reply]
Agreed! I will make this change. Given that the committee will have 6 elected seats (3 primary + 3 alternate), I would suggest no more than 2 be from the same company

**Reply by dims** [reply]
+1 to `max 2 from the same company`
